### PR TITLE
[Android] Add support of self managed call

### PR DESCRIPTION
**Pull request by luxiliu**
The changes in this pull request is to support self managed ConnectionService. 

Screen Shot 2020-11-03 at 11 00 35 am

Example app is updated to show how to use it.

1.  Set `selfManaged` of `IOptions` to true.
2.  Add `RNCallKeep.addEventListener('showIncomingCall', youMethodToShowIncomingCall)`
3.  Implement your own logic In `youMethodToShowIncomingCall`

### CODE REVIEW

**Comment by sboily** [comment]
Hello, it could be great to have a README fo this new feature for explaining what is it and how to use it.

**Comment by luxiliu** [comment]
> Hello, it could be great to have a README fo this new feature for explaining what is it and how to use it.

👌 Comment is updated above 👆

**Comment by danwhite-ipc** [comment]
Does this solution handle all app state scenarios? ie when the app is in the background or dead?

**Comment by luxiliu** [comment]
> Does this solution handle all app state scenarios? ie when the app is in the background or dead?

I'm not sure. I think this change is not related to whether app state is background or dead. It just implements https://developer.android.com/reference/android/telecom/Connection#onShowIncomingCallUi(), and expose a callback for RN app.

**Comment by wilau2** [comment]
> Does this solution handle all app state scenarios? ie when the app is in the background or dead?
It probably does not handle the scenarios you have in mind.. 
You'll have to make sure, that your incomingCall activity have the window flags: 
```
        getWindow().addFlags(WindowManager.LayoutParams.FLAG_SHOW_WHEN_LOCKED | WindowManager.LayoutParams.FLAG_TURN_SCREEN_ON);
```
But in order for this to work, you will have to make sure that the activity that you set those flags on will be destroyed and created again when needed. 

This is the only way that we've been able to have the incoming call screen always display on top of the keyguard.

**Review by MaliahRajan — APPROVED** [review]

**Comment by sboily** [comment]
We need to make some tests before merge this PR, but if there is not objections, i think we could merge it after the tests. @manuquentin and @danjenkins any objections about this new features?

**Comment by marcato15** [comment]
And progress on this?

**Comment by luxiliu** [comment]
How's the tests going?

**Comment by sboily** [comment]
Hello, sorry we don't have find time to start the test... We'll do as soon as possible.

**Comment by luxiliu** [comment]
Closing as #395 is created

**Pull request closed without merging**
